src/node/blockstorage.cpp

```diff
@@ -527,7 +527,7 @@ void BlockManager::FlushUndoFile(int block_file, bool finalize)
 {
     FlatFilePos undo_pos_old(block_file, m_blockfile_info[block_file].nUndoSize);
     if (!UndoFileSeq().Flush(undo_pos_old, finalize)) {
-        AbortNode("Flushing undo file to disk failed. This is likely the result of an I/O error.");
+        AbortNode("Flushing undo file to disk failed. This is likely the result of an I/O error.", m_opts.init_error_callback);
```

Comment:
These new AbortNode calls seem more awkward. And they aren't really future-proof since `AbortNode` calls the global `StartShutdown` function which is specific to the node and will have to be changed for the kernel. Also the name "abort node" doesn't make sense in the kernel context if the kernel application isn't a bitcoin node. Would suggest adding a `FatalError` notification to complement `InitError` in `ChainstateNotifications` class suggested previously:

```c++
class ChainstateNotifications
{
    ...
    virtual void FatalError(const bilingual_str& error_message, const bilingual_str abort_message={}) {}
};
```

Other kernel clients could customize `FatalError` but the node implementation would call `SetMiscWarning(error_message)` and `InitError(abort_message or "A fatal internal error occurred, see debug.log for details")` and `StartShutdown()` like `AbortNode` function does currently, to keep behavior unchanged.